crates/evm/evm/src/executors/fuzz/mod.rs

```diff
@@ -98,7 +98,19 @@ impl FuzzedExecutor {
         let max_traces_to_collect = std::cmp::max(1, self.config.gas_report_samples) as usize;
         let show_logs = self.config.show_logs;
 
+        // Start a timer if timeout is set
+        let start_time = self.config.timeout_secs.map(|timeout| {
+            (std::time::Instant::now(), std::time::Duration::from_secs(timeout))
+        });
+
         let run_result = self.runner.clone().run(&strategy, |calldata| {
+            // Check if the timeout has been reached
+            if let Some((start_time, timeout)) = start_time {
+                if start_time.elapsed() > timeout {
+                    return Err(TestCaseError::fail("Timeout reached"))
```

Comment:
I used an error here but in theory it could also go the other way and just set some sort of shared boolean flag and return `Ok()`. You would have to handle it in the `Ok()` block down below instead of the `Err(TestError::Fail(reason, _))` block. Will happily take any feedback on that. 

Only thing about returning a failure is that this triggers proptest to save the results, not sure if that's important at all. You get some message about proptest doing stuff.

Reply:
yeah, indeed that the main reason this feature wasn't implemented yet, proptest PR to interrupt test was proposed here https://github.com/proptest-rs/proptest/pull/521

Ref https://github.com/foundry-rs/foundry/issues/990

Reply:
Do you see any issue with just failing like I do in the PR or returning `Ok()`? Locally it seems to work fine but not sure if there's something I'm missing about the behavior here.

Reply:
This is sort of a weak timeout in the sense that it just stops new jobs from being created rather than interrupting any running jobs. But it's better than nothing and works most of the time.

Reply:
I think this makes sense for the use case where we only want to timeout tests if they take longer than desired (was under the impression you want time based campaigns where you set smth like run tests for one day). We need to fail it, otherwise the tests will continue to be executed until all runs completed.

Reply:
Yeah basically my idea to achieve this in practice is to set some insane number of fuzz runs and then have it timeout after a day. Should work?

Reply:
that should work (though the max runs is capped by proptest at u32 max value, that is 4294967295).
One more thing to consider here is that tests are parallelized using rayon, so if you have let's say 2 cores available and want to run 4 tests with huge number of fuzz and timeout then there will be 2 tests run until timeout and only after other 2 tests will be picked up for execution (so if you want to timeout after 1h the entire test run will actually timeout in 2h).